src/classes/AccountsSelector.cls

```diff
@@ -0,0 +1,87 @@
+public class AccountsSelector extends fflib_SObjectSelector {
+    
+    List<String> selectFields = new List<String>{'Name', 'BillingCity', 'BillingState', 'BillingCountry'};
+
+    public List<Schema.SObjectField> getSObjectFieldList() {
+        return new List<Schema.SObjectField> {
+            Account.Name,
+            Account.Description,
+            Account.AnnualRevenue,
+            Account.Id};
+    }
+
+    public AccountsSelector()
+	{
+		super(false);
+	}
+	
+	public AccountsSelector(Boolean includeFieldSetFields)
+	{
+		super(includeFieldSetFields);
+	}
+    
+    
+    public Schema.SObjectType getSObjectType() {
+        return Account.sObjectType;
+    }
+
+    public List<Account> selectById(Set<ID> idSet) {
+        return (List<Account>) selectSObjectsById(idSet);
+    }
+    
+    public List<Account> selectAccountWithFilter(String selectedBillingCity, String selectedBillingState, String selectedBillingCountry)
+    {
+        String queryString = createQuery(selectedBillingCity,selectedBillingState,selectedBillingCountry);
```

Comment:
Returned String doesn't have limit clause then how could you assure that not more than 100 records returned through this query.

Reply:
We have another method with Limit clause in it.